src/sctools/platform.py

```diff
@@ -935,15 +935,16 @@ def _make_tag_generators(
         barcode_args = {'fastq_files': r1}
 
         if i1:
-            barcode_args['embedded_barcodes'] = [cls.sample_barcode]
-            tag_generators.append(fastq.EmbeddedBarcodeGenerator(**barcode_args))
+            sample_barcode_args = dict(barcode_args)
```

Comment:
Can you explain why this change was needed in the PR description? Otherwise I agree with the tests and their assertions.

Reply:
OK